cmd/av/stack.go

```diff
@@ -0,0 +1,173 @@
+}
+
+var stackBranchFlags struct {
+	// The parent branch to base the new branch off.
+	// By default, this is the current branch.
+	Parent string
+}
+var stackBranchCmd = &cobra.Command{
+	Use:   "branch <branch name>",
+	Short: "create a new stacked branch",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return errors.New("branch name is required")
+		}
+		return errors.New("unimplemented")
+	},
+}
+
+var stackSyncFlags struct {
+	// Set the parent of the current branch to this branch.
+	// This effectively re-roots the stack on a new parent.
+	Parent string
+	// If set, incorporate changes from the trunk (repo base branch) into the stack.
+	// Only valid if synchronizing the root of a stack.
+	// This effectively re-roots the stack on the latest commit from the trunk.
+	Trunk bool
+	// If set, do not push to GitHub.
+	NoPush bool
+	// If set, we're continuing a previous sync.
+	// TODO:
+	// 	 we might not actually need this, we can probably detect that
+	//   a sync needs to be completed automagically and do the right thing.
+	Continue bool
+}
+var stackSyncCmd = &cobra.Command{
+	Use:   "sync",
+	Short: "synchronize all stacked branches after the current branch",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return errors.New("unimplemented")
+	},
+}
+
+var stackTreeFlags struct {
+	// Print the stack starting at this branch.
+	// If not set, we start at the base branch of the repository.
+	Root string
+	// Only recurse up-to the given depth.
+	Depth int
+}
+var stackTreeCmd = &cobra.Command{
+	Use:   "tree",
+	Short: "show the tree of stacked branches",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return errors.New("unimplemented")
+	},
+}
+
+var stackNextCmd = &cobra.Command{
+	Use:   "next <n>",
+	Short: "checkout the next branch in the stack",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var n int = 1
+		if len(args) == 1 {
+			var err error
+			n, err = strconv.Atoi(args[0])
+			if err != nil {
+				return errors.New("invalid number")
+			}
+		} else if len(args) > 1 {
+			_ = cmd.Usage()
+			return errors.New("too many arguments")
+		}
+
+		if n <= 0 {
+			return errors.New("invalid number (must be >= 1)")
+		}
+
+		return errors.New("unimplemented")
+	},
+}
+
+var stackPrevCmd = &cobra.Command{
+	Use:   "prev <n>",
+	Short: "checkout the previous branch in the stack",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var n int = 1
+		if len(args) == 1 {
+			var err error
+			n, err = strconv.Atoi(args[0])
+			if err != nil {
+				return errors.Wrap(err, "invalid number")
+			}
+		} else if len(args) > 1 {
+			_ = cmd.Usage()
+			return errors.New("too many arguments")
+		}
+
+		if n <= 0 {
+			return errors.New("invalid number (must be >= 1)")
+		}
+
+		return errors.New("unimplemented")
+	},
+}
+
+var stackStackStatus = &cobra.Command{
+	Use:   "status",
+	Short: "show the status of the stack",
+	Long:  `Show the status of the stack.`,
```

Comment:
why have a long desc? I don't see elsewhere.

Reply:
Whoops, will fix.

FYI, the short is what shows up in the `Available Commands` section:
```
managed stacked pull requests

Usage:
  av stack [command]

Available Commands:
  branch      create a new stacked branch
  next        checkout the next branch in the stack
  prev        checkout the previous branch in the stack
  status      show the status of the stack
  sync        synchronize all stacked branches after the current branch
  tree        show the tree of stacked branches

Flags:
  -h, --help   help for stack

Global Flags:
      --debug   enable verbose debug logging

Use "av stack [command] --help" for more information about a command.
```

The long is what shows up in the help text.
```
go run ./cmd/av stack status --help
Show the status of the stack.

Usage:
  av stack status [flags]

Flags:
  -h, --help   help for status

Global Flags:
      --debug   enable verbose debug logging
```